aten/src/TH/generic/THTensor.cpp

```diff
@@ -351,12 +351,17 @@ void THTensor_(select)(THTensor *self, THTensor *src, int dimension, int64_t sli
 
   THTensor_(set)(self, src);
   THTensor_(narrow)(self, NULL, dimension, sliceIndex, 1);
+
+  std::vector<int64_t> newSize(self->dim()-1);
+  std::vector<int64_t> newStride(self->dim()-1);
   for(d = dimension; d < self->dim()-1; d++)
   {
-    self->set_size(d, self->size(d+1));
-    self->set_stride(d, self->stride(d+1));
+    auto self_size = THTensor_sizeLegacyNoScalars(self, d+1);
```

Comment:
Hmm, I think it's OK to use `self->size(d+1)` here, instead of the legacy one, because it is not permissible to call select on a 0-dim tensor (the only case when this matters.)

Reply:
I see, thank you @ezyang ! I'll fix it!

Reply:
pushed new changes